src/mpyl/steps/deploy/dagster.py

```diff
@@ -94,19 +96,51 @@ def execute(self, step_input: Input) -> Output:
             return self.__evaluate_results(dagster_deploy_results)
 
         name_suffix = get_name_suffix(properties)
+        release_name = convert_to_helm_release_name(
+            step_input.project.name, name_suffix
+        )
+
+        builder = ChartBuilder(step_input)
+
+        sealed_secrets_manifest = builder.to_sealed_secrets()
+        # adjust manifest name to suffixed-release name
+        sealed_secrets_manifest.metadata.name = release_name
+        sealed_secrets_for_target = list(
+            filter(
+                lambda v: v.get_value(properties.target) is not None,
+                step_input.project.deployment.properties.sealed_secret
+                if step_input.project.deployment
+                else [],
+            )
+        )
+        sealed_secrets_as_env_var = [
+            V1EnvVar(
+                name=e.key,
+                value_from=V1EnvVarSource(
+                    secret_key_ref=V1SecretKeySelector(
+                        key=e.key, name=release_name, optional=False
+                    )
+                ),
+            )
+            for e in sealed_secrets_for_target
+        ]
```

Comment:
I would potentially move most of this logic into the `ChartBuilder` (so we localize the k8s client related objects there)